digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/SpeedLimitService.scala

```diff
@@ -269,7 +269,32 @@ class SpeedLimitService(eventbus: DigiroadEventBus, vvhClient: VVHClient, roadLi
     */
   def updateValues(ids: Seq[Long], value: Int, username: String, municipalityValidation: Int => Unit): Seq[Long] = {
     withDynTransaction {
-      ids.flatMap(dao.updateSpeedLimitValue(_, value, username, municipalityValidation))
+      ids.flatMap(updateSpeedLimitValue(_, value, username, municipalityValidation))
+    }
+  }
+
+  /**
+    * Create new speed limit when value received from UI changes and expire the old one. Used by SpeeedLimitsService.updateValues.
+    */
+  def updateSpeedLimitValue(id: Long, value: Int, username: String, municipalityValidation: Int => Unit): Option[Long] = {
+    def validateMunicipalities(vvhLinks: Seq[(Long, Double, Seq[Point], Int)]): Unit = {
+      vvhLinks.foreach(vvhLink => municipalityValidation(vvhLink._4))
+    }
+
+    validateMunicipalities(dao.getLinksWithLengthFromVVH(20, id))
+
+    //Get all data from the speedLimit to update
+    val speedLimit = dao.getPersistedSpeedLimit(id).map(toSpeedLimit).get
+
```

Comment:
I belive we don't need to do the convertion "toSpeedlimit", the persistedSpeedLimit have all the information